### ISSUE TITLE
Unexpected error ErrorCodes.ConnectionCloseForced: The connection was inactive for more than the allowed 240000 milliseconds and is closed by container

### ISSUE THREAD

**Issue by saramonteiro** [Azure/azure-iot-sdk-python#947]
# Context

- **OS and version used:** <VERSION> (Windows 10 Enterprise)
- **Python version:** <VERSION> (python 3.10.1)
- **pip version:** <VERSION> (pip 21.2.4)
- **list of installed packages:** <VERSION>
 Package            Version
azure-core         1.22.0
azure-iot-hub      2.6.0
certifi            2021.10.8
charset-normalizer 2.0.11
idna               3.3
isodate            0.6.1
msrest             0.6.21
oauthlib           3.2.0
pip                21.2.4
requests           2.27.1
requests-oauthlib  1.3.1
setuptools         58.1.0
six                1.16.0
uamqp              1.5.1
urllib3            1.26.8
- **cloned repo:** <VERSION> ( commit dd1cc544f2531a55edb563668803bb12ef6db1a2 | branch main )

## Description of the issue

The issue is that the connection between application SDK and IoT Hub automatically drops after 240 s if no message is sent during this interval.

## Code sample exhibiting the issue

<details>
<summary>Sample Based on iothub_registry_manager_c2d_sample.py </summary>

``` python

# -------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for
# license information.
# --------------------------------------------------------------------------

import sys
import os
import msrest
from azure.iot.hub import IoTHubRegistryManager
import time

connection_str = 'MY SERVICE CS'
device_id = 'Device0'

try:
    # Create IoTHubRegistryManager
    registry_manager = IoTHubRegistryManager.from_connection_string(connection_str)
    print("Conn String: {0}".format(connection_str))

    # Send Message To Device
    send_message = "Sending c2d message 1"
    registry_manager.send_c2d_message(device_id, send_message)

    for i in range(241):
        time.sleep(1)
        print(i, "s")

    # Send 2nd message - This message is never sent
    send_message = "Sending 2nd c2d message 1"
    registry_manager.send_c2d_message(device_id, send_message)

    print("Closing sample")

except msrest.exceptions.HttpOperationError as ex:
    print("HttpOperationError error {0}".format(ex.response.text))
except Exception as ex:
    print("Unexpected error {0}".format(ex))
except KeyboardInterrupt:
    print("{} stopped".format(__file__))
finally:
    print("{} finished".format(__file__))

```
</details>

## Console log of the issue

After the loop of +240 s, the following error is thrown:

`Unexpected error ErrorCodes.ConnectionCloseForced: The connection was inactive for more than the allowed 240000 milliseconds and is closed by container 'DeviceGateway_<a long number here>'.`

## May be related:

https://github.com/Azure/azure-sdk-for-python/issues/9917

**Comment by christian-e-petersen** [comment]
is there a fix for this? I am facing the same error running on aks

**Comment by cartertinney** [comment]
@christian-e-petersen @saramonteiro

We are working to get this addressed internally, our apologies for the delay. We will update this issue when we have more information.

**Comment by cartertinney** [comment]
Fixed in #955, will be included in next major release

**Comment by 98christianp** [comment]
Hey @cartertinney, just an update from my side. I was using version 5.8.0a1 of azure-eventhub. Changing to version 5.7.0 solved my issues. Thanks for the update :)
